src/transformers/tokenization_utils_fast.py

```diff
@@ -555,3 +562,50 @@ def _save_pretrained(
             file_names = file_names + (tokenizer_file,)
 
         return file_names
+
+    def train_new_from_iterator(self, text_iterator, vocab_size, special_tokens=None, **kwargs):
+        tokenizer_json = json.loads(self._tokenizer.to_str())
+        # Remove added tokens for now (uses IDs of tokens)
+        added_tokens = tokenizer_json.pop("added_tokens")
+        # Remove post processor for now (uses IDs of tokens)
+        post_processor = tokenizer_json.pop("post_processor")
+        # Remove vocab
+        if tokenizer_json["model"]["type"] == "BPE":
+            tokenizer_json["model"]["vocab"] = {}
+            tokenizer_json["model"]["merges"] = []
+        elif tokenizer_json["model"]["type"] == "Unigram":
+            if tokenizer_json["model"]["unk_id"] is not None:
+                unk_id = tokenizer_json["model"]["unk_id"]
+                tokenizer_json["model"]["unk_id"] = 0
+                tokenizer_json["model"]["vocab"] = [tokenizer_json["model"]["vocab"][unk_id]]
+        elif tokenizer_json["model"]["type"] == "WordPiece":
+            tokenizer_json["model"]["vocab"] = {}
+
+        tokenizer = TokenizerFast.from_str(json.dumps(tokenizer_json))
+
+        # Get the special tokens from the current tokenizer if none are specified.
+        if special_tokens is None:
+            special_tokens = []
+            for added_token in added_tokens:
+                if tokenizer_json["model"]["type"] != "Unigram":
+                    special = added_token.pop("special")
+                    if not special:
+                        continue
+                    _ = added_token.pop("id")
+                special_tokens.append(AddedToken(**added_token))
+
+        trainer_class = MODEL_TO_TRAINER_MAPPING[tokenizer_json["model"]["type"]]
```

Comment:
You can also get and modify the trainer directly with
```python
trainer = tokenizer.model.get_trainer()
trainer.vocab_size = vocab_size
trainer.special_tokens = special_tokens
```
Another advantage of doing this, is that it should work directly when we'll be able to get the trainer with the parameters that were used for the initial training.

Reply:
Can all kwargs accepted by the init be passed to the `get_trainer` method? The idea was to let the user customize all of them.